File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Users/Views/CallGradientView/CallGradientView.m
//
//  ChatGradientView.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 13.06.2020.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "CallGradientView.h"

@interface CallGradientView()
//MARK: - Properties
@property (strong, nonatomic) CAGradientLayer *gradientLayer;
@end

@implementation CallGradientView
//MARK: - Life Cycle
- (instancetype)initWithCoder:(NSCoder *)coder {
    self = [super initWithCoder:coder];
    if (self) {
        self.gradientLayer = [CAGradientLayer layer];
        self.gradientLayer.startPoint = CGPointZero;
        self.gradientLayer.endPoint = CGPointMake(0.0f, 1.0f);
        [self applyGradient];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];

    [self updateGradientFrame];
}

//MARK: - Public Methods
- (void)setupGradientWithFirstColor:(UIColor *)firstColor andSecondColor:(UIColor *)secondColor {
    self.gradientLayer.colors = @[(id)firstColor.CGColor, (id)secondColor.CGColor];
    [self applyGradient];
}

//MARK: - Internal Methods
- (void)applyGradient {
    self.gradientLayer.frame = self.bounds;
    [self.layer insertSublayer:self.gradientLayer atIndex:0];
}

- (void)updateGradientFrame {
    self.gradientLayer.frame = self.bounds;
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Calls/Views/CallGradientView/CallGradientView.swift
//
//  CallGradientView.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 5/26/20.
//  Copyright © 2020 QuickBlox. All rights reserved.
//

import UIKit

class CallGradientView: UIView {
    //MARK: - Properties
    lazy var gradientLayer: CAGradientLayer = {
        let layer = CAGradientLayer()
        layer.startPoint = CGPoint.zero
        layer.endPoint = CGPoint(x: 0, y: 1)
        return layer
    }()
    
    //MARK: - Life Cycle
    override func awakeFromNib() {
        super.awakeFromNib()
        
        applyGradient()
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        updateGradientFrame()
    }
    
    //MARK: - Public Methods
    func setupGradient(firstColor: UIColor, secondColor: UIColor) {
        gradientLayer.colors = [firstColor.cgColor, secondColor.cgColor]
        applyGradient()
    }
    
    //MARK: - Private Methods
    private func applyGradient() {
        gradientLayer.frame = bounds
        layer.insertSublayer(gradientLayer, at: 0)
    }
    
    private func updateGradientFrame() {
        gradientLayer.frame = bounds
    }
}
